app/scripts/controllers/rif/rns/register/index.js

```diff
@@ -25,63 +26,60 @@ export default class RnsRegister extends RnsJsDelegate {
    * Make a request for registration on the domainName for an amount of years.
    * @param domainName the Domain to register.
    * @param yearsToRegister the amount of years to register.
-   * @returns {Promise<string>} secret to use on the finishRegistration operation.
+   * @returns {Promise<string>} commitment that helps is to check if you can finish registration or you still have to wait.
    */
   requestRegistration (domainName, yearsToRegister) {
-    const domainHash = web3Utils.sha3(domainName);
+    const cleanDomainName = this.cleanDomainFromRskPrefix(domainName);
+    const domainHash = web3Utils.sha3(cleanDomainName);
     const secret = generateRandomSecret();
     return new Promise((resolve, reject) => {
-      this.call(this.fifsContractInstance, 'makeCommitment', [domainHash, this.address, secret])
-        .then(commitment => {
-          console.debug('Commitment received', commitment);
-          this.sendTransaction(this.fifsContractInstance, 'commit', [commitment])
-            .then(result => {
-              console.debug('Commitment committed', result);
-              const state = this.getStoreState();
-              if (!state.register[this.address]) {
-                state.register[this.address] = {};
-              }
-              state.register[this.address].domainRegister = {
-                secret,
-                domainName,
-                yearsToRegister,
-              };
-              this.updateStoreState(state);
+      this.getDomainCost(cleanDomainName, yearsToRegister)
+        .then(rifCost => {
+          this.call(this.fifsAddrRegistrarInstance, 'makeCommitment', [domainHash, this.address, secret])
+            .then(commitment => {
+              console.debug('Commitment received', commitment);
+              this.send(this.fifsAddrRegistrarInstance, 'commit', [commitment])
+                .then(result => {
+                  console.debug('Commitment committed', result);
```

Comment:
What do you think about moving this to` async/await` instead of promise `then`?

Reply:
I preffer promise handling but if we need to change it i can do that.